include/engine/hdf5/HDF5WriterP.h

```diff
@@ -0,0 +1,224 @@
+/*
+ * HDF5WriterP.h
+ *
+ *  Created on: March 20, 2017
+ *      Author: Junmin
+ */
+
+#ifndef HDF5_WRITER_P_H_
+#define HDF5_WRITER_P_H_
+
+#include <iostream> //std::cout must be removed, only used for hello example
+#include <unistd.h> //sleep must be removed
+
+#include "core/Engine.h"
+#include "format/BP1Writer.h"
+
+// supported capsules
+#include "capsule/heap/STLVector.h"
+
+#include "ADIOS_MPI.h"
+
+#include "hdf5.h"
+
+namespace adios {
+
+typedef struct {
+  double _re; /*real part*/
+  double _im; /*imaginary part*/
+} ADIOS2_Complex_Double;
+
+typedef struct {
+  float _re; // real
+  float _im; // imaginary part
+} ADIOS2_Complex_Float;
+
+// using ADIOS2_Complex_Float = std::complex<float>;
+
+typedef struct {
+  long double _re; /*real part*/
+  long double _im; /*imaginary part*/
+} ADIOS2_Complex_LongDouble;
+
```

Comment:
Rather than creating your own structs for complex float, double, and long double, why can't you just use std::complex?